packages/interactivity/src/init.ts

```diff
@@ -45,13 +77,7 @@ export const init = async () => {
 	} );
 
 	for ( const node of nodes ) {
-		if ( ! hydratedIslands.has( node ) ) {
-			await splitTask();
-			const fragment = getRegionRootFragment( node );
-			const vdom = toVdom( node );
-			initialVdom.set( node, vdom );
-			await splitTask();
-			hydrate( vdom, fragment );
-		}
+		pendingNodes.add( node );
+		intersectionObserver.observe( node );
```

Comment:
I think it might be a good idea to keep the `hydratedIslands.has` check here? No need to even observe nodes that are already hydrated, e.g. if `init` was called multiple times. Unlikely, but probably a good safety net to have.
```suggestion
		if ( ! hydratedIslands.has( node ) ) {
			pendingNodes.add( node );
			intersectionObserver.observe( node );
		}
```

Reply:
The `hydratedIslands.has()` check was moved to the IntersectionObserver callback. It could be added here as well, but after abd37478b5e64375f4ca63a3bcabd6c0e5d01c6e that would complicate things a bit since it could be that the `nodes.length` number would end up being larger than the number of nodes actually observed, meaning the `IntersectionObserver` would never get disconnected.